src/client-agent/config.c

```diff
@@ -128,14 +128,26 @@ cJSON *getLabelsConfig(void) {
 
     unsigned int i;
     cJSON *root = cJSON_CreateObject();
-    cJSON *labels = cJSON_CreateObject();
+    cJSON *label_s = cJSON_CreateArray();
+    cJSON *label = cJSON_CreateObject();
+    cJSON *labels = cJSON_CreateArray();
 
     if (agt->labels) {
         for (i=0;agt->labels[i].key;i++) {
-            cJSON_AddStringToObject(labels,agt->labels[i].key,agt->labels[i].value);
+            cJSON *label_k = cJSON_CreateObject();
+            cJSON_AddStringToObject(label_k,"value", agt->labels[i].value);
+            cJSON_AddStringToObject(label_k,"key", agt->labels[i].key);
+            if (agt->labels[i].flags.hidden)
+                cJSON_AddStringToObject(label_k,"hidden", "yes");
+            if (agt->labels[i].flags.system)
```

Comment:
As per a conversation with @chemamartinez, system flags should not appear.
Configuration is being read here, and system flags are not part of the configuration. 
See https://documentation.wazuh.com/3.9/user-manual/reference/ossec-conf/labels.html